### PR TITLE
Add support for multiple configurations

### PR DESCRIPTION
**Pull request by chris-wood**
cc @bwesterb for 👀

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -161,11 +162,21 @@ func main() {
...
+	// From the primary configuration ID, create a key ID for the legacy configuration that old
+	// clients will use for obtaining configuration material. This will eventually be removed once all
+	// clients have been updated to support the primary configuration ID.
+	legacyConfigID := uint8((configID - 10) % 255)
+	seed[len(seed)-1] ^= 0xFF
+	legacyConfig, err := ohttp.NewConfigFromSeed(legacyConfigID, hpke.KEM_X25519_HKDF_SHA256, hpke.KDF_HKDF_SHA256, hpke.AEAD_AES128GCM, seed)
+	if err != nil {
+		log.Fatalf("Failed to create legacy gateway configuration from seed: %s", err)
+	}
+
```
Comment (chris-wood) [inline main.go]: This is the hack.

File: main.go [hunk 2]
```diff
@@ -161,11 +163,21 @@ func main() {
...
 
+	// From the primary configuration ID, create a key ID for the legacy configuration that old
+	// clients will use for obtaining configuration material. This will eventually be removed once all
+	// clients have been updated to support the primary configuration ID.
+	legacyConfigID := uint8((configID - 128) % 255)
+	seed[len(seed)-1] ^= 0xFF
+	legacyConfig, err := ohttp.NewConfigFromSeed(legacyConfigID, hpke.KEM_X25519_HKDF_SHA256, hpke.KDF_HKDF_SHA256, hpke.AEAD_AES128GCM, seed)
+	if err != nil {
+		log.Fatalf("Failed to create legacy gateway configuration from seed: %s", err)
+	}
```
Comment (chris-wood) [inline main.go:179]: This is the hack.

File: go.mod [hunk 3]
```diff
@@ -4,17 +4,13 @@ go 1.17
 
 require (
 	github.com/DataDog/datadog-go/v5 v5.1.1
```
Comment (bwesterb) [inline go.mod:6]: Did you `go mod tidy`?

File: main.go [hunk 4]
```diff
@@ -161,11 +163,21 @@ func main() {
...
+	config, err := ohttp.NewConfigFromSeed(configID, hpke.KEM_X25519_KYBER768_DRAFT00, hpke.KDF_HKDF_SHA256, hpke.AEAD_AES128GCM, seed)
 	if err != nil {
 		log.Fatalf("Failed to create gateway configuration from seed: %s", err)
 	}
 
+	// From the primary configuration ID, create a key ID for the legacy configuration that old
+	// clients will use for obtaining configuration material. This will eventually be removed once all
+	// clients have been updated to support the primary configuration ID.
+	legacyConfigID := uint8((configID - 128) % 255)
+	seed[len(seed)-1] ^= 0xFF
```
Comment (bwesterb) [inline main.go:175]: Why not the same seed?

**Review by bwesterb — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (chris-wood) [reply go.mod:6]: Yep 👍

File: main.go (hunk 4 above)
Comment (chris-wood) [reply main.go:175]: It feels wrong to use the same randomness for two different purposes, so I flipped some bits here.

**Pull request merged**